lib/msal-core/src/AuthError.ts

```diff
@@ -0,0 +1,240 @@
+/**
+  * Copyright (c) Microsoft Corporation
+  *  All Rights Reserved
+  *  MIT License
+  *
+  * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  * software and associated documentation files (the "Software"), to deal in the Software
+  * without restriction, including without limitation the rights to use, copy, modify,
+  * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  * permit persons to whom the Software is furnished to do so, subject to the following
+  * conditions:
+  *
+  * The above copyright notice and this permission notice shall be
+  * included in all copies or substantial portions of the Software.
+  *
+  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+  * OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
+  * OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+  */
+
+import { Constants } from "./Constants";
+import { ErrorMessage } from "./ErrorMessage";
+import { Utils } from "./Utils";
+/**
+* @hidden
+*
+* General error class thrown by the MSAL.js library.
+*/
+export class AuthError extends Error {
+    error: string;
+    errorDesc: string;
+
+    constructor(error: string, detailMessage: string) {
```

Comment:
error is a little generic and non descript.
What do you think of `errorCode`?

Reply:
error is the name of the var coming from the server, so we decided to keep it uniform

Reply:
code and message are a little more descriptive though.  error indicates error object, code indicates a string const representation of an error, and message indicates the details of that given code

Reply:
@manoj-rath @sameerag Looking at other libraries, here are their error object signatures:
Android: errorCode, errorMessage
iOS: code, desc
C#: StatusCode, ExceptionDetail

I am going to use errorCode and errorMessage similar to Android since it seems like there isn't much uniformity across MSAL currently.

Reply:
+1

Reply:
Will change this to display errorCode and errorMessage in line with Android.